cli/flox-rust-sdk/src/models/lockfile.rs

```diff
@@ -126,6 +122,135 @@ impl LockedManifest {
         Ok(ContainerBuilder::new(container_builder_path))
     }
 
+    pub fn read_from_file(path: &CanonicalPath) -> Result<Self, LockedManifestError> {
+        let contents = fs::read(path).map_err(LockedManifestError::ReadLockfile)?;
+        serde_json::from_slice(&contents).map_err(LockedManifestError::ParseLockfile)
+    }
+
+    pub fn check_lockfile(
+        path: &CanonicalPath,
+    ) -> Result<Vec<LockfileCheckWarning>, LockedManifestError> {
+        let mut pkgdb_cmd = Command::new(Path::new(&*PKGDB_BIN));
+        pkgdb_cmd
+            .args(["manifest", "check"])
+            .arg("--lockfile")
+            .arg(path.as_os_str());
+
+        debug!("checking lockfile with command: {}", pkgdb_cmd.display());
+
+        let value = call_pkgdb(pkgdb_cmd).map_err(LockedManifestError::CheckLockfile)?;
+        let warnings: Vec<LockfileCheckWarning> =
+            serde_json::from_value(value).map_err(LockedManifestError::ParseCheckWarnings)?;
+
+        Ok(warnings)
+    }
+}
+
+impl ToString for LockedManifest {
+    fn to_string(&self) -> String {
+        serde_json::json!(self).to_string()
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct LockedManifestCatalog {
+    #[serde(rename = "lockfile-version")]
+    version: Version<1>,
+    /// original manifest that was locked
+    manifest: TypedManifestCatalog,
+    /// locked groups
+    groups: Vec<LockedGroup>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+struct LockedGroup {
+    /// name of the group
+    name: String,
+    /// system this group provides packages for
+    system: System,
+    /// [CatalogPage] that was selected to fulfill this group
+    ///
+    /// If resolution of a group provides multiple pages,
+    /// a single page is selected based on cross group constraints.
+    /// By default this is the latest page that provides packages
+    /// for all requested systems.
+    page: CatalogPage,
```

Comment:
note: `packages` in a `CatalogPage` may be empty, and if it is, we'll have to perform another request to populate it. Whereas here I think the semantics we want are that it's never empty. I'm not sure it's worth using two different types, but I thought it was at least worth mentioning